### ISSUE TITLE
Enhancement: More supported sound formats in Chart Editor

### ISSUE THREAD

**Issue by amyspark-ng** [FunkinCrew/Funkin#2432]
## What is your suggestion, and why should it be implemented?
Wish chart editor could take different file formats for making charts, like mp3, wav etc, instead of only ogg

**Comment by Lasercar** [comment]
Can the game actually use different those different formats for music/songs?

**Comment by AbnormalPoof** [comment]
> Can the game actually use different those different formats for music/songs?

Yes, Flixel supports `.mp3`, `.ogg`, and `.wav`. It's just a matter of adapting the game to support multiple file extensions.

**Comment by Lasercar** [comment]
> > Can the game actually use different those different formats for music/songs?
> 
> Yes, Flixel supports `.mp3`, `.ogg`, and `.wav`. It's just a matter of adapting the game to support multiple file extensions.

Hmm, I’ll see if it’s as simple as allowing those the other supported file types to be selected in the file browser that is opened (it probably won’t be though). I’ll investigate later.

**Comment by Lasercar** [comment]
Hmm, wavs work, but mp3s don't, weirdly.

https://github.com/user-attachments/assets/8a8846db-436f-420f-a46b-100490adfd11
(thought it'd be funny if I loaded a wav from a completely different (and very old) game)

I had to allow waveforms to be null so I could make it not throw a crash as trying to load mp3 files ended up doing that.

There's also the issue that most paths are hard set to use the Constants.EXT_SOUND
https://github.com/FunkinCrew/Funkin/blob/b74df06f48400bdb3c23dd27ad69a24b92cfe625/source/funkin/util/Constants.hx#L517
so even if it did properly load mp3s, both still wouldn't be actually usable ingame, because it'll be looking for only ogg or mp3 files, depending on whether it's a web build or not.
